application_handlers.go

```diff
@@ -233,3 +233,61 @@ func SourceListApplications(c echo.Context) error {
 
 	return c.JSON(http.StatusOK, util.CollectionResponse(out, c.Request(), int(count), limit, offset))
 }
+
+// ApplicationPause pauses a given application by setting its "paused_at" column to "now()".
+func ApplicationPause(c echo.Context) error {
+	applicationId, err := strconv.ParseInt(c.Param("id"), 10, 64)
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, util.ErrorDoc(err.Error(), "400"))
+	}
+
+	applicationDao, err := getApplicationDao(c)
+	if err != nil {
+		return err
+	}
+
+	application, err := applicationDao.Pause(applicationId)
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, util.ErrorDoc(err.Error(), "400"))
```

Comment:
Sorry if I am wrong. When applicationDao.Pause() returns Not Found error, then I think here it will be replaced by Bad Request. Same situation in ApplicationResume(). Same in source_handler.go.